packages/edit-site/src/index.js

```diff
@@ -58,7 +58,9 @@ export function initialize( id, settings ) {
 
 	registerCoreBlocks();
 	if ( process.env.GUTENBERG_PHASE === 2 ) {
-		__experimentalRegisterExperimentalCoreBlocks( settings );
+		__experimentalRegisterExperimentalCoreBlocks( settings, {
+			postType: 'wp_template',
```

Comment:
I feel like this could be a little confusing.  `postType` makes a lot of sense in edit-post, but thinking of this for edit-site it seems a bit more funky.  Is the post type always `wp_template` in edit-site?  Even when we are selecting specific pages or posts?  If we are going to hardcode a value here maybe it should just be something like 'edit_site'.

Perhaps something like `editorType`, `editorEntity`, or some better name I cant think of atm instead of `postType`.
So for edit post we would pass in `{ editorType: postType }`.
For site editor `{ editorType: 'site_editor' }`, and add that as a 3rd valid option in the array checked in the ternary?

Or maybe just a comment here explaining why we are going with `wp_template` as the hardcoded value, that the block wont be registered if this isn't passed as `wp_template` or `wp_template_part`. 🤷‍♀️

Reply:
> I feel like this could be a little confusing.  `postType` makes a lot of sense in edit-post, but thinking of this for edit-site it seems a bit more funky.  Is the post type always `wp_template` in edit-site?  Even when we are selecting specific pages or posts? If we are going to hardcode a value here maybe it should just be something like 'edit_site'.

Yeah, it will be `wp_template` in those cases, too -- it's still the `wp_template` CPT we're editing. (However, it can also be `wp_template_part`, if we're editing a template part.)

A few factors come into play here. One is that we're only setting `postType` upon initialization -- we don't re-initialize upon switching between templates or template parts. We know that the Site Editor initially always shows a template (and not a template part) -- the home page template -- so we hard-code to `wp_template`. It just so happens that template parts allow for the same block types (and disallow Post Content).

> Perhaps something like `editorType`, `editorEntity`, or some better name I cant think of atm instead of `postType`.
So for edit post we would pass in `{ editorType: postType }`.
For site editor `{ editorType: 'site_editor' }`, and add that as a 3rd valid option in the array checked in the ternary?

Yep, which brings us to the question whether `postType` is the right argument to base these allow/deny lists on anyway. The reason for choosing `postType` (but really post context -- the second argument to `__experimentalRegisterExperimentalCoreBlocks` is an object that would also allow for additional fields such as e.g. post ID etc) is that it aligns with the [`allowed_block_types` filter](https://developer.wordpress.org/reference/hooks/allowed_block_types/), which accepts a post object. Arguably, our client-side filter should be aligned with the server-side one, API-wise, since they're semantically doing the same thing.

However, it's arguable that a post object simply isn't the completely right fit anyway, and that we should replace it with something like `editorType`. Coincidentally, @TimothyBJacobs arrived at a similar [conclusion](https://github.com/WordPress/gutenberg/pull/24076#issuecomment-661808248) in a separate issue (about providing the list of allowed blocks via the REST API). So maybe it's time to consider that, and maybe even deprecate/replace the current `allowed_block_types` filter (cc/ @youknowriad) We would lose the ability to filter on a per-post basis, but OTOH, a post object might also not be the best fit for some (edge?) cases that Timothy lists in that discussion.

> Or maybe just a comment here explaining why we are going with `wp_template` as the hardcoded value, and some block(s) wont be registered if this isn't passed as `wp_template` or `wp_template_part`. 🤷‍♀️ 

Agree, writing this reply I realized there's a lot of considerations that have gone into this.